units/patch_intent.go

```diff
@@ -368,7 +368,27 @@ func (j *patchIntentProcessor) finishPatch(ctx context.Context, patchDoc *patch.
 	}
 
 	if patchDoc.IsChild() && !j.intent.ShouldFinalizePatch() {
-		// TODO subscribe on parent patch outcome
+		parentStatus := j.intent.GetParentStatus()
```

Comment:
An alternative to adding this to the intent interface could be to do this part of the processing in `buildTriggerPatchDoc()`. That function already type asserts the intent so you can access its fields.

Reply:
I will need this function for [4453](https://github.com/evergreen-ci/evergreen/pull/4453). Is there still a benefit to moving it?

Reply:
If we go with [this idea](https://github.com/evergreen-ci/evergreen/pull/4453#discussion_r587767749) we could do without it there as well.

Reply:
After querying for the intent, we need to access intent.ParentStatus, which we can't access directly.